src/dos/cdrom_image.cpp

```diff
@@ -1001,13 +1015,28 @@ bool CDROM_Interface_Image::ReadSectorsHost(void *buffer, bool raw, unsigned lon
 	return success;
 }
 
+void CDROM_Interface_Image::PlayNextAudioTrack()
+{
+	if (currentTrackNumber >= tracks.size()) {
+		StopAudio();
+		return;
+	}
+
+	// track number is index + 1 so current track number == next track index
+	const auto &next_track = tracks[currentTrackNumber];
+	assert(next_track.number == currentTrackNumber + 1);
```

Comment:
Hmm, why don't we store `currentTrackIndex` instead internally, then things would become more obvious, perhaps?

Reply:
Internally where? I don't know what you mean by that.

Reply:
Oh, number to index. That's the part I didn't see. Yeah I can do that.

Reply:
done